Add Flink 1.19.1 release

hlteoh37 opened the pull request:
- Add Flink 1.19.1 release blog announcement scheduled for 2024-06-14.
- Add Flink 1.19.1 release to release archive.
- Change Flink 1.19 to point to Flink 1.19.1

hlteoh37 committed:
Add Flink 1.19.1 release

jeyhunkarimov reviewed (approved):
LGTM

1996fanrui commented on docs/content/posts/2024-06-14-release-1.19.1.md:
```diff
@@ -0,0 +1,171 @@
+</li>
+</ul>
+
+<h2>        New Feature
+</h2>
+<ul>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-34633'>FLINK-34633</a>] -         Support unnesting array constants
```
In general, the patch version shouldn't include the new Feature.

After I checked, i think the `Fix Version` of FLINK-34633 should be marked as `1.20.0` instead of `1.19.1`. Let's follow it in FLINK-34633.

hlteoh37 replied:
Removed. Thank you for checking!

1996fanrui commented on docs/content/posts/2024-06-14-release-1.19.1.md:
```diff
@@ -0,0 +1,171 @@
+The Apache Flink Community is pleased to announce the first bug fix release of the Flink 1.19 series.
+
+This release includes 44 bug fixes, vulnerability fixes, and minor improvements for Flink 1.19.
+Below you will find a list of all bugfixes and improvements (excluding improvements to the build infrastructure and build stability). For a complete list of all changes see:
```
As the release Announcement mentioned: `excluding improvements to the build infrastructure and build stability`.

The release Announcement is `user oriented`, we should remove all JIRAs related to `build infrastructure and build stability` in `Release Notes ` part, such as:
- RetryingExecutorTest. testDiscardOnTimeout failed on azure
- TableSourceITCase#testTableHintWithLogicalTableScanReuse sometimes fails with result mismatch 
- AsyncSinkWriterThrottlingTest failed with Illegal mailbox thread
- etc

hlteoh37 replied:
Removed the below JIRAs from release notes. Will add as comment on PR

### Subtask
- [FLINK-33816](https://issues.apache.org/jira/browse/FLINK-33816) -         SourceStreamTaskTest.testTriggeringStopWithSavepointWithDrain failed due async checkpoint triggering not being completed 
- [FLINK-34707](https://issues.apache.org/jira/browse/FLINK-34707) -         Update japicmp configuration
- [FLINK-34712](https://issues.apache.org/jira/browse/FLINK-34712) -         Update reference data for Migration Tests

### Bug
- [FLINK-26515](https://issues.apache.org/jira/browse/FLINK-26515) -         RetryingExecutorTest. testDiscardOnTimeout failed on azure
- [FLINK-29114](https://issues.apache.org/jira/browse/FLINK-29114) -         TableSourceITCase#testTableHintWithLogicalTableScanReuse sometimes fails with result mismatch 
- [FLINK-31472](https://issues.apache.org/jira/browse/FLINK-31472) -         AsyncSinkWriterThrottlingTest failed with Illegal mailbox thread
- [FLINK-34324](https://issues.apache.org/jira/browse/FLINK-34324) -         s3_setup is called in test_file_sink.sh even if the common_s3.sh is not sourced
- [FLINK-34569](https://issues.apache.org/jira/browse/FLINK-34569) -         &#39;Streaming File Sink s3 end-to-end test&#39; failed
- [FLINK-34571](https://issues.apache.org/jira/browse/FLINK-34571) -         SortMergeResultPartitionReadSchedulerTest.testOnReadBufferRequestError failed due an assertion
- [FLINK-34617](https://issues.apache.org/jira/browse/FLINK-34617) -         Correct the Javadoc of org.apache.flink.api.common.time.Time
- [FLINK-34622](https://issues.apache.org/jira/browse/FLINK-34622) -         Typo of execution_mode configuration name in Chinese document
- [FLINK-34933](https://issues.apache.org/jira/browse/FLINK-34933) -         JobMasterServiceLeadershipRunnerTest#testResultFutureCompletionOfOutdatedLeaderIsIgnored isn&#39;t implemented properly
- [FLINK-35000](https://issues.apache.org/jira/browse/FLINK-35000) -         PullRequest template doesn&#39;t use the correct format to refer to the testing code convention
- [FLINK-35383](https://issues.apache.org/jira/browse/FLINK-35383) -         Update compatibility matrix to include 1.19 release
- [FLINK-35526](https://issues.apache.org/jira/browse/FLINK-35526) -         Remove deprecated stedolan/jq Docker image from Flink e2e tests

### Technical Debt
- [FLINK-34409](https://issues.apache.org/jira/browse/FLINK-34409) -         Increase test coverage for AdaptiveScheduler
- [FLINK-34897](https://issues.apache.org/jira/browse/FLINK-34897) -         JobMasterServiceLeadershipRunnerTest#testJobMasterServiceLeadershipRunnerCloseWhenElectionServiceGrantLeaderShip needs to be enabled again

1996fanrui commented on docs/content/posts/2024-06-14-release-1.19.1.md:
```diff
@@ -0,0 +1,171 @@
+
+        Release Notes - Flink - Version 1.19.1
+
+<h2>        Sub-task
```
It's better to remove the subtask part, and move these JIRAs to bug or Improvement part.

Sub-task means it's a subtask JIRA. But for user side, each jira is bugfix or Improvement or new feature.

hlteoh37 replied:
Ok - removed. All subtasks are not relevant to users

1996fanrui reviewed (commented):
Thanks @hlteoh37 for the release!

I left some comments, please take a look in your free time, thanks~

hlteoh37 commented:
Thanks @1996fanrui  - great comments! I've addressed them.

I've removed the below JIRAs from the release notes as they related to build infrastructure:

### Subtask
- [FLINK-33816](https://issues.apache.org/jira/browse/FLINK-33816) -         SourceStreamTaskTest.testTriggeringStopWithSavepointWithDrain failed due async checkpoint triggering not being completed 
- [FLINK-34707](https://issues.apache.org/jira/browse/FLINK-34707) -         Update japicmp configuration
- [FLINK-34712](https://issues.apache.org/jira/browse/FLINK-34712) -         Update reference data for Migration Tests

### Bug
- [FLINK-26515](https://issues.apache.org/jira/browse/FLINK-26515) -         RetryingExecutorTest. testDiscardOnTimeout failed on azure
- [FLINK-29114](https://issues.apache.org/jira/browse/FLINK-29114) -         TableSourceITCase#testTableHintWithLogicalTableScanReuse sometimes fails with result mismatch 
- [FLINK-31472](https://issues.apache.org/jira/browse/FLINK-31472) -         AsyncSinkWriterThrottlingTest failed with Illegal mailbox thread
- [FLINK-34324](https://issues.apache.org/jira/browse/FLINK-34324) -         s3_setup is called in test_file_sink.sh even if the common_s3.sh is not sourced
- [FLINK-34569](https://issues.apache.org/jira/browse/FLINK-34569) -         &#39;Streaming File Sink s3 end-to-end test&#39; failed
- [FLINK-34571](https://issues.apache.org/jira/browse/FLINK-34571) -         SortMergeResultPartitionReadSchedulerTest.testOnReadBufferRequestError failed due an assertion
- [FLINK-34617](https://issues.apache.org/jira/browse/FLINK-34617) -         Correct the Javadoc of org.apache.flink.api.common.time.Time
- [FLINK-34622](https://issues.apache.org/jira/browse/FLINK-34622) -         Typo of execution_mode configuration name in Chinese document
- [FLINK-34933](https://issues.apache.org/jira/browse/FLINK-34933) -         JobMasterServiceLeadershipRunnerTest#testResultFutureCompletionOfOutdatedLeaderIsIgnored isn&#39;t implemented properly
- [FLINK-35000](https://issues.apache.org/jira/browse/FLINK-35000) -         PullRequest template doesn&#39;t use the correct format to refer to the testing code convention
- [FLINK-35383](https://issues.apache.org/jira/browse/FLINK-35383) -         Update compatibility matrix to include 1.19 release
- [FLINK-35526](https://issues.apache.org/jira/browse/FLINK-35526) -         Remove deprecated stedolan/jq Docker image from Flink e2e tests

### Technical Debt
- [FLINK-34409](https://issues.apache.org/jira/browse/FLINK-34409) -         Increase test coverage for AdaptiveScheduler
- [FLINK-34897](https://issues.apache.org/jira/browse/FLINK-34897) -         JobMasterServiceLeadershipRunnerTest#testJobMasterServiceLeadershipRunnerCloseWhenElectionServiceGrantLeaderShip needs to be enabled again

1996fanrui reviewed (approved):
Thanks @hlteoh37 for the update!

LGTM

leonardBang commented on docs/content/posts/2024-06-14-release-1.19.1.md:
```diff
@@ -0,0 +1,128 @@
+---
+title:  "Apache Flink 1.19.1 Release Announcement"
+date: "2024-06-14T00:00:00.000Z"
+aliases:
+- /news/2024/06/14/release-1.19.1.html
```
tips: please update the date when fire the release.

hlteoh37 replied:
Released artifacts on 06-14 so I will keep this date!
